gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -127,21 +133,28 @@ public TrinoQueryProperties(
         this.isNewQuerySubmission = isNewQuerySubmission;
         this.isQueryParsingSuccessful = isQueryParsingSuccessful;
         isClientsUseV2Format = false;
+        maxBodySize = -1;
     }
 
     public TrinoQueryProperties(HttpServletRequest request, RequestAnalyzerConfig config)
     {
-        isClientsUseV2Format = config.isClientsUseV2Format();
+        this(request, config.isClientsUseV2Format(), config.getMaxBodySize());
+    }
+
+    public TrinoQueryProperties(HttpServletRequest request, boolean isClientsUseV2Format, int maxBodySize)
+    {
+        this.isClientsUseV2Format = isClientsUseV2Format;
```

Comment:
Add `requireNonNull`.

Reply:
this is a `boolean` primitive, so the null check is redundant

Reply:
Sorry, I forgot to mention the variable name. We should add the check to `HttpServletRequest request`